### PR TITLE
Spanish translation of the file debugging.mdx

### PR DESCRIPTION
**Pull request by SimplyJuanjo**
# What does this PR do?

Fixes #[15947](https://github.com/huggingface/transformers/issues/15947)
Adds the Spanish version of [debugging.mdx](https://github.com/huggingface/transformers/blob/master/docs/source/en/debugging.mdx) to [transformers/docs/source/es](https://github.com/huggingface/transformers/tree/main/docs/source/es)

Also found one typo error in the original doc, so I fixed it also.

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

## Who can review?

@omarespejel @osanseviero  @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing the typo and the translation! You need to add your new file in the spanish Table of Contents (the file named `_toctree.yml`)

Would also like @osanseviero to have a look before we merge :-)

File: docs/source/es/debugging.mdx [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Debugging
+
+## Debug de problemas de Network multi-GPU
+
+Cuando entrene o infiera con `DistributedDataParallel` y varias GPUs, si encuentras problemas de intercomunicación entre procesos y/o nodos, puedes usar el siguiente script para diagnosticar problemas de red.
```
Comment (osanseviero) [inline docs/source/es/debugging.mdx:17]:
```suggestion
Cuando entrenas o infieres con `DistributedDataParallel` y varias GPUs, si encuentras problemas de intercomunicación entre procesos y/o nodos, puedes usar el siguiente script para diagnosticar problemas de red.
```

File: docs/source/es/debugging.mdx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+<Tip>
+
+Esta función está disponible actualmente sólo para PyTorch.
+
+</Tip>
+
+<Tip>
+
+Para el entrenamiento multi-GPU requiere DDP (`torch.distributed.launch`).
```
Comment (osanseviero) [inline docs/source/es/debugging.mdx:53]:
```suggestion
Para el entrenamiento multi-GPU, requiere DDP (`torch.distributed.launch`).
```

File: docs/source/es/debugging.mdx [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+1.79e-06 4.65e+00 input[0]
+3.18e-04 6.27e+04 output
+                  encoder.block.2.layer.1.dropout Dropout
+3.18e-04 6.27e+04 input[0]
+0.00e+00      inf output
+```
+
+El output del ejemplo se ha recortado en el centro por razones de brevedad.
+
+La segunda columna muestra el valor del elemento más grande en términos absolutos, por lo que si se observas con detenimiento los últimos fotogramas,
```
Comment (osanseviero) [inline docs/source/es/debugging.mdx:130]:
```suggestion
La segunda columna muestra el valor del elemento más grande en términos absolutos, por lo que si observas con detenimiento los últimos fotogramas,
```

File: docs/source/es/debugging.mdx [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+0.00e+00      inf output
+```
+
+El último frame informa para la función `Dropout.forward` con la primera entrada para el único input y la segunda para el
+único output. Puedes ver que fue llamada desde un atributo `dropout` dentro de la clase `DenseReluDense`. Podemos ver
+que ocurrió durante la primera capa, del segundo block, durante el primer batch. Por último, el mayor absoluto
+elementos de input fue `6.27e+04` y el mismo para el output fue `inf`.
+
+Puedes ver aquí, que `T5DenseGatedGeluDense.forward` resultó en output activations, cuyo valor máximo absoluto fue
+alrededor de 62.7K, que está muy cerca del límite máximo de fp16 de 64K. En el siguiente frame tenemos `Dropout` que renormaliza
```
Comment (osanseviero) [inline docs/source/es/debugging.mdx:184]:
```suggestion
alrededor de 62.7K, que está muy cerca del límite máximo de fp16 de 64K. En el siguiente frame tenemos `Dropout`, el cual renormaliza
```

File: docs/source/es/debugging.mdx [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+        detect_overflow(forwarded_states, "after layer_norm")
+        forwarded_states = self.DenseReluDense(forwarded_states)
+        detect_overflow(forwarded_states, "after DenseReluDense")
+        return hidden_states + self.dropout(forwarded_states)
+```
+
+Puedes ver que hemos añadido 2 de estos y ahora se trackea si `inf` o `nan` para `forwarded_states` fue detectado
+en algún punto intermedio.
+
+De hecho, el detector ya informa de esto porque cada una de las calls en el ejemplo anterior es un `nn.Module`, pero
```
Comment (osanseviero) [inline docs/source/es/debugging.mdx:264]:
```suggestion
De hecho, el detector ya informa de esto porque cada una de las llamadas en el ejemplo anterior es un `nn.Module`, pero
```

**Review by osanseviero — APPROVED** [review]
Thanks a lot for the contribution! :fire:

**Comment by sgugger** [comment]
Can you just add the new file to the TOC of the Spanish doc? (in `transformers/docs/source/es/_toctree.yml`)

**Comment by sgugger** [comment]
All good, thanks!

**Pull request merged**